gapic-common/lib/gapic/grpc/service_stub.rb

```diff
@@ -49,22 +54,43 @@ class ServiceStub
     #     provided as a `::GRPC::Core::Channel`.)
     # @param interceptors [Array<::GRPC::ClientInterceptor>] An array of {::GRPC::ClientInterceptor} objects that will
     #   be used for intercepting calls before they are executed Interceptors are an EXPERIMENTAL API.
+    # @param channel_pool_config [::Gapic::ServiceStub:ChannelPool::Configuration] The configuration for channel
+    #     pool. (This argument is ignored when `credentials` is provided as a `::GRPC::Core::Channel`.)
     #
-    def initialize grpc_stub_class, endpoint:, credentials:, channel_args: nil, interceptors: nil
+    def initialize grpc_stub_class, endpoint:, credentials:, channel_args: nil,
+                   interceptors: nil, channel_pool_config: nil
       raise ArgumentError, "grpc_stub_class is required" if grpc_stub_class.nil?
       raise ArgumentError, "endpoint is required" if endpoint.nil?
       raise ArgumentError, "credentials is required" if credentials.nil?
 
       channel_args = Hash channel_args
       interceptors = Array interceptors
 
+      if channel_pool_config && channel_pool_config.channel_count > 1 && !(credentials.is_a? ::GRPC::Core::Channel)
+        create_channel_pool grpc_stub_class, endpoint: endpoint, credentials: credentials,
+                            channel_args: channel_args, interceptors: interceptors,
+                            channel_pool_config: channel_pool_config
+      else
+        create_grpc_stub grpc_stub_class, endpoint: endpoint, credentials: credentials,
+                         channel_args: channel_args, interceptors: interceptors
+      end
+    end
+
+    def create_channel_pool grpc_stub_class, endpoint:, credentials:, channel_args: nil,
+                            interceptors: nil, channel_pool_config: nil
+      @channel_pool = ChannelPool.new grpc_stub_class, endpoint: endpoint, credentials: credentials,
```

Comment:
Currently we're setting either `@channel_pool` or `@grpc_stub` and leaving the other unset. It's better to set the other variable explicitly to nil. Unset instance variables do have an implicit value of nil, but 2.x versions of Ruby will still emit warnings when you access an unset instance variable so it's better to set them explicitly.

Reply:
Updated. Explicitly setting both `grpc_stub` and `channel_pool` to `nil`.